src/class_singleCleaner.py

```diff
@@ -171,7 +171,7 @@ def resendPubkeyRequest(self, address):
             # is already present and will not do the POW and send the message
             # because it assumes that it has already done it recently.
             del state.neededPubkeys[address]
-        except:
+        except:  # noqa:E722
```

Comment:
```except KeyError:```

Reply:
**This is just an explanation**

If we somehow manage to trigger a `RuntimeError`, this will raise an exception, but ignoring it will also cause trouble down the line as the key won't be requested. But because this function is called inside custom garbage collection, it shouldn't happen. If it happens, we'll have BM crashing with `RuntimeError` and we'll know what's happening, as opposed to silently not requesting a key.

Reply:
Add a separate handler for RuntimeError, which will print a log entry with "warning" severity, saying that "Can't remove key in \_\_file\_\_ \_\_class\_\_ \_\_line\_\_, this shouldn't happen but if it does it prevents a pubkey from being requested".

Reply:
Updated.

Reply:
Oh, perhaps you can just add `exc_info=True` to the `logger.warning()` call in the later except clause? Than you will have much more info than  `__file__` `__class__` `__line__`

Reply:
Updated